xarray/coding/cftimeindex.py

```diff
@@ -541,14 +541,29 @@ def shift(self, n, freq):
         >>> index.shift(1, "M")
         CFTimeIndex([2000-02-29 00:00:00],
                     dtype='object', length=1, calendar='gregorian', freq=None)
+        >>> index.shift(1.5, "D")
+        CFTimeIndex([2000-02-01 12:00:00],
+                    dtype='object', length=1, calendar='gregorian', freq=None)
         """
         from .cftime_offsets import to_offset
```

Comment:
`to_offset` is only needed when freq is a `str`. Moving this inside that check will give a slight performance improvement for timedelta cases.